hack/verify-reports.sh

```diff
@@ -0,0 +1,109 @@
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+error() {
+  echo "ERROR: $*" 1>&2
+}
+
+info() {
+  echo "INFO: $*" 1>&2
+}
+
+# Check if the provided Gateway API version is greater than or equal to v1.1.0
+check_ge_v1.1.0() {
+    local version=$1
+    local minimum_version="v1.1.0"
+
+    # Normalize versions to remove leading 'v' and compare using sort -V
+    if [[ $(echo -e "${version#v}\n${minimum_version#v}" | sort -V | head -n1) == "${minimum_version#v}" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Check if the report fields are valid according to the rules defined in https://github.com/kubernetes-sigs/gateway-api/blob/release-1.1/conformance/reports/README.md
+check_report_fields() {
+    local report=$1
+    local expected_gateway_api_version=$2
+
+    # Check if the implementation version is a valid semver
+    local version=$(yq eval '.implementation.version' "$report")
+    local gateway_api_version=$(yq eval '.gatewayAPIVersion' "$report")
+    local gateway_api_channel=$(yq eval '.gatewayAPIChannel' "$report")
+    local mode=$(yq eval '.mode' "$report")
+    local version_pattern="^v?[0-9]+\.[0-9]+(\.[0-9]+)?$"
+
+    if [[ ! $version =~ $version_pattern ]]; then
+        error "$report version does not match the semver pattern"
+        EXIT_VALUE=1
+    fi
+    if [[ ${gateway_api_version} != ${expected_gateway_api_version} ]]; then
+        error "$report gatewayAPIVersion does not match Gateway API version folder"
+        EXIT_VALUE=1
+    fi
+    if [[ $gateway_api_channel != "standard" && $gateway_api_channel != "experimental" ]]; then
+        error "$report gatewayAPIChannel is neither standard nor experimental"
+        EXIT_VALUE=1
+    fi
+    if [[ $mode == "" ]]; then
+        error "$report mode must be set"
+        EXIT_VALUE=1
+    fi
+}
+
+REPORTS_DIR=$(dirname "${BASH_SOURCE}")/../conformance/reports
+# Regex to match the report file name pattern defined in https://github.com/kubernetes-sigs/gateway-api/blob/release-1.1/conformance/reports/README.md#how-this-folder-is-structured
+FILE_NAMING_PATTERN="^(standard|experimental)-v?[0-9]+\.[0-9]+(\.[0-9]+)?-[^-]+-report\.yaml$"
+EXIT_VALUE=0
+
+for dir in ${REPORTS_DIR}/*
+do
+    element="${dir##*/}"
+    if check_ge_v1.1.0 "${element}"; then
+        if [[ -d "${dir}" ]]; then
+            gateway_api_version="${element}"
+            for implementation_dir in ${dir}/*
+            do
+                implementation=$(basename "${implementation_dir}")
+                info "Checking ${implementation} project directory for Gateway API version ${gateway_api_version}"
+            
+                if [[ -f "${implementation_dir}/README.md" ]]; then
+                # Check if the README.md has broken links
+                    docker run -v $(readlink -f "$implementation_dir"):/${implementation}:ro --rm -i ghcr.io/tcort/markdown-link-check:stable /${implementation}/README.md
```

Comment:
How does this utility work? Can/should we use this for all of our docs? I think some of our links are only valid with mkdocs-material context, would that be able to understand this?

Reply:
> all the internal links should already be checked when building the documentation

Not that I know of?

> I think this is out of scope for this PR, would you agree with me opening an issue to improve the verify.sh script to fix them?

Agree it's out of scope for this PR, but a follow up would be great.

Reply:
👍 I've created https://github.com/kubernetes-sigs/gateway-api/issues/3258 for this. For what concerns internal links with mkdocs, I asked confirmation directly to the community: https://matrix.to/#/!agmMfyRRWLhOnjmeFl:gitter.im/$sc_nZQIzGCVAu3hLKj0hzLLCeroYTEbn2Wfc12kUfGU?via=gitter.im&via=matrix.org